lib/configurators/databases/aws.go

```diff
@@ -0,0 +1,639 @@
+	if a.helper == nil {
+		return ctx, trace.BadParameter("policy helper not initialized")
+	}
+
+	policyArn, ok := ctx.Value(policyArnKey).(string)
+	if !ok {
+		return ctx, trace.BadParameter("policy ARN not present")
+	}
+
+	boundaryArn, ok := ctx.Value(policyBoundaryArnKey).(string)
+	if !ok {
+		return ctx, trace.BadParameter("policy boundary not present")
+	}
+
+	err := a.helper.AttachPolicy(ctx, policyArn, a.target)
+	if err != nil {
+		return ctx, trace.Wrap(err)
+	}
+
+	err = a.helper.AttachPolicyBoundary(ctx, boundaryArn, a.target)
+	if err != nil {
+		return ctx, trace.Wrap(err)
+	}
+
+	return ctx, nil
+}
+
+// getPolicyHelper returns a `PolicyHelper` following the rules:
+// * If the `configurator` has a `policyHelper` configured, return it;
+// * If the `identity` or `session` are `nil`, return a `nil` `PolicyHelper`;
+// * Otherwise, create the `PolicyHelper` and save it to the `configurator`, and
+//   return it;
+func (a *awsConfigurator) getPolicyHelper(identity awslib.Identity, session *awssession.Session) awslib.PolicyHelper {
+	if a.policyHelper != nil {
+		return a.policyHelper
+	}
+
+	if identity == nil || session == nil {
+		return nil
+	}
+
+	a.policyHelper = awslib.NewPolicyHelper(identity.GetAccountID(), iam.New(session))
+	return a.policyHelper
+}
+
+// build generates the policy documents and store them into the
+// `awsConfigurator` struct.
+func (a *awsConfigurator) build() error {
+	var (
+		err      error
+		session  *awssession.Session
+		identity awslib.Identity
+	)
+
+	// When running the command in manual mode, we want to have zero dependency
+	// with AWS configurations (like awscli or environment variables), so that
+	// the user can run this command and generate the instructions without any
+	// pre-requisite.
+	if !a.flags.Manual {
+		session, err = a.getAWSSession()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		stsClient, err := a.getAWSSTSClient()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		identity, err = awslib.GetIdentityWithClient(context.Background(), stsClient)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	accountID := "*"
+	if identity != nil {
+		accountID = identity.GetAccountID()
+	}
+
+	// Define the target and target type.
+	target, err := policiesTarget(a.flags, accountID, identity)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Generate policies.
+	policyDocument, err := buildPolicyDocument(a.flags, a.fileConfig, accountID, target)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	policyBoundaryDocument, err := buildPolicyBoundaryDocument(a.flags, a.fileConfig, accountID, target)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	formattedPolicy, err := formatPolicyDocument(policyDocument)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	formattedPolicyBoundary, err := formatPolicyDocument(policyBoundaryDocument)
+	if err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
It is kept outside so that neither we need an `awsPolicyCreator/awsPoliciesAttacher` constructors nor the `Details` function needs to return an error. For me, it makes sense to keep this close to the policy build since if there is an issue with it, the root cause is "close by".